torch/csrc/autograd/custom_function.cpp

```diff
@@ -100,4 +100,50 @@ variable_list _wrap_outputs(const variable_list &input_vars,
   return outputs;
 }
 
+void AutogradContext::save_for_backward(const variable_list &to_save) {
+  saved_variables.clear();
```

Comment:
This design is a bit iffy. In the original version, `_save_variables` (btw, remember the comment I made about cross-referencing the implementations before and after? Please do that here too!), this function is not actually a public user visible method; it is called after application of the python function to read out the `to_save` member field (which the user actually wrote) and save them in a more durable form. But here, you've defined it as a user visible function, which ostensibly is called by implementations of AutogradContext. But now there is a problem: you haven't wrapped the output variable yet, so `(var.grad_fn().get() == ptr.get())` will always be false! That's no good.